lib/dstate/interface.go

```diff
@@ -263,6 +263,14 @@ type ChannelState struct {
 	ThreadMetadata   *discordgo.ThreadMetadata `json:"thread_metadata"`
 
 	PermissionOverwrites []discordgo.PermissionOverwrite `json:"permission_overwrites"`
+
+	AvailableTags []discordgo.ForumTag `json:"available_tags"`
+	AppliedTags   []int64              `json:"applied_tags"`
+
+	DefaultReactionEmoji          discordgo.ForumDefaultReaction `json:"default_reaction_emoji"`
+	DefaultThreadRateLimitPerUser int                            `json:"default_thread_rate_limit_per_user"`
+	DefaultSortOrder              *discordgo.ForumSortOrderType  `json:"default_sort_order"`
+	DefaultForumLayout            discordgo.ForumLayout          `json:"default_forum_layout"`
```

Comment:
I think, for IsPrivate() method few lines down, `|| c.Type == discordgo.ChannelTypeGuildPrivateThread` needs to be added to `if`

Reply:
Does GuildPrivateThread sit on the same level as DM privacy?

Reply:
looking at how this is handled in context functions, well it is a badly named field. Channel Type 12 says yeah it's private as well. so better not change this.